### PR TITLE
feat: ci workflow v2

### PR DESCRIPTION
**Pull request by liya2017**
### What problem does this PR solve?
Improve CI Workflows. Spilt `ci.yaml` into 17 child workflows. Those workflows triggered by push/pull_request
Support skip jobs run , like `ci-runs-only: [unit_tests, linters]`  just run unit_tests and linters.
Support skip os run, like `ci-runs-on: [windows, ubuntu]` will skip runs on macos 

### Check List 

Tests

- Unit test
- Integration test

### Release note

```release-note
None
```

### CODE REVIEW

File: .github/actions/ci_prologue/action.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: 'ci-composite-action'
+description: 'Greet someone'
```
Comment (doitian) [inline .github/actions/ci_prologue/action.yml:2]: Change this to some reasonable description.

File: .github/actions/ci_prologue/ci_prologue.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+set -e
+fun_run_os(){
+ runs_on="$1"
```
Comment (doitian) [inline .github/actions/ci_prologue/ci_prologue.sh:4]: Weird indentation. Please use 2 spaces consistently.

File: .github/actions/ci_prologue/ci_prologue.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  if [[ "$ci_uses_self_runner" == "true" ]] || [[ "$GITHUB_REPOSITORY" == "nervosnetwork/ckb" ]];then
+    runner_label='self-hosted-ci-ubuntu-20.04'
+  else
+    runner_label='ubuntu-20.04'
+  fi
+  echo "runner_label is "$runner_label
+  echo "::set-output name=runner_label::$runner_label"
+}
+
+if [[ $GITHUB_EVENT_NAME == "push" ]];then
```
Comment (doitian) [inline .github/actions/ci_prologue/ci_prologue.sh:78]: Use 2 spaces for indentation.

File: devtools/ci/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#
```
Comment (doitian) [inline devtools/ci/requirements.txt:1]: It seems there are no python scripts now. Could this file be deleted?

File: devtools/ci/ci_main.sh [hunk 5]
```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+set -u
+set +e
+echo $GITHUB_WORKSPACE
+CARGO_TARGET_DIR=${CARGO_TARGET_DIR:-"$GITHUB_WORKSPACE/target"}
+case $GITHUB_WORKFLOW in
+	ci_linters*)
```
Comment (doitian) [inline devtools/ci/ci_main.sh:7]: Use consistent indentation in all shell files.

**Review by doitian — CHANGES_REQUESTED** [review]

**Review by doitian — APPROVED** [review]

**Review by quake — APPROVED** [review]

**Comment by quake** [comment]
bors r=quake,doitian

File: devtools/ci/ci_main.sh [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+is_self_runner=`echo $RUNNER_LABEL | awk -F '-' '{print $1}'`
+if [[ $is_self_runner == "self" ]];then
+  CARGO_TARGET_DIR=$GITHUB_WORKSPACE/../target
+fi
+CARGO_TARGET_DIR=${CARGO_TARGET_DIR:-"$GITHUB_WORKSPACE/target"}
+case $GITHUB_WORKFLOW in
+  ci_linters*)
+    echo "Hellow ci_linters!"
+    cargo fmt --version ||  rustup component add rustfmt
+    argo clippy --version ||  rustup component add clippy
```
Comment (chanhsu001) [inline devtools/ci/ci_main.sh:13]: typo issue, please change from "argo" to "cargo",  and "||" hide the error.

**Pull request merged**